scheduler/cloudformation/cloudformation.go

```diff
@@ -257,22 +261,109 @@ func (s *Scheduler) submit(ctx context.Context, tx *sql.Tx, app *scheduler.App,
 			Parameters: parameters,
 			// TODO: Update Go client
 			// Tags:         tags,
-		}, done, ss); err != nil {
+		}, output, ss); err != nil {
 			return err
 		}
 	} else {
 		return fmt.Errorf("error describing stack: %v", err)
 	}
 
 	if ss != nil {
-		if err := <-done; err != nil {
-			return err
+		if err := s.waitUntilStable(ctx, output, ss); err != nil {
+			logger.Warn(ctx, fmt.Sprintf("error waiting for submit to stabilize: %v", err))
 		}
 	}
+	return nil
+}
 
+func (s *Scheduler) waitUntilStable(ctx context.Context, o chan stackOperationOutput, ss scheduler.StatusStream) error {
+	output := <-o
+	if output.err != nil || output.stack == nil {
+		return output.err
+	}
+
+	deployments, err := deploymentsToWatch(output.stack)
+	if err != nil {
+		return err
+	}
+	deploymentStatuses := s.waitForDeploymentsToStabilize(ctx, deployments)
+	for status := range deploymentStatuses {
+		scheduler.Publish(ctx, ss, fmt.Sprintf("Service %s became %s", status.deployment.service, status))
+	}
+	// TODO publish notification to empire
 	return nil
 }
 
+type deploymentStatus struct {
+	deployment *ecsDeployment
+	status     string
+}
+
+func (d *deploymentStatus) String() string {
+	return d.status
+}
+
+type deploymentIndex struct {
+	index      int
+	deployment *ecsDeployment
+}
+
+func (s *Scheduler) waitForDeploymentsToStabilize(ctx context.Context, deployments []*ecsDeployment) <-chan deploymentStatus {
+	ch := make(chan deploymentStatus)
+
+	go func(deployments []*ecsDeployment) {
+		for len(deployments) > 0 {
+			var serviceIDs []*string
+			arnToDeployment := make(map[string]*deploymentIndex)
+			for i, d := range deployments {
+				serviceIDs = append(serviceIDs, aws.String(d.serviceID))
+				arnToDeployment[d.serviceARN] = &deploymentIndex{i, d}
+			}
+
+			status, err := s.ecs.DescribeServices(&ecs.DescribeServicesInput{
+				Cluster:  aws.String(s.Cluster),
+				Services: serviceIDs,
```

Comment:
I think you can just use the `serviceARN` here and remove the `serviceID` field.